src/test/integration/IntegrationBase.t.sol

```diff
@@ -0,0 +1,367 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "forge-std/Test.sol";
+
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import "src/test/integration/IntegrationDeployer.t.sol";
+import "src/test/integration/Global.t.sol";
+import "src/test/integration/users/User.sol";
+
+abstract contract IntegrationBase is IntegrationDeployer {
+
+    /**
+     * Gen/Init methods:
+     */
+
+    /**
+     * @dev Create a new user according to configured random variants.
+     * This user is ready to deposit into some strategies and has some underlying token balances
+     */
+    function _newStaker() internal returns (User, IStrategy[] memory, uint[] memory) {
+        (User staker, IStrategy[] memory strategies, uint[] memory tokenBalances) = _randUser();
+        
+        assert_HasUnderlyingTokenBalances(staker, strategies, tokenBalances, "_newStaker: failed to award token balances");
+
+        return (staker, strategies, tokenBalances);
+    }
+
+    function _newOperator() internal returns (User, IStrategy[] memory, uint[] memory) {
+        (User operator, IStrategy[] memory strategies, uint[] memory tokenBalances) = _randUser();
+        
+        operator.registerAsOperator();
+        // TODO - deposit into strats
+
+        assert_HasUnderlyingTokenBalances(operator, strategies, tokenBalances, "_newOperator: failed to award token balances");
+        assertTrue(delegationManager.isOperator(address(operator)), "_newOperator: operator should be registered");
```

Comment:
could have a variant that changes the order of registering as an operator and depositing?
we can also add an `assert_HasOperatorShares` at the end 😄

Reply:
What we could do is update `User.registerAsOperator` to register -> deposit by default, and create like a `User_DepositFirst`-type of user variant... but i'm not sure where it's useful outside of this specific case.

I think we should talk about what additional variants we want to see during the call today!

For now I'm going to keep a static ordering for this, and add `assert_Snap_AddedOperatorShares` + `assert_Snap_AddedStakerShares` as checks

Reply:
[6ef08a6](https://github.com/Layr-Labs/eigenlayer-contracts/pull/335/commits/6ef08a676fe70389631c4627a9aa2d825a660884)